cli/loader/src/lib.rs

```diff
@@ -304,16 +305,21 @@ impl Loader {
     }
 
     fn language_for_id(&self, id: usize) -> Result<Language> {
-        let (path, language) = &self.languages_by_id[id];
+        let (path, language, external_cfg) = &self.languages_by_id[id];
         language
             .get_or_try_init(|| {
                 let src_path = path.join("src");
-                self.load_language_at_path(&src_path, &src_path)
+                self.load_language_at_path(&src_path, &src_path, external_cfg)
             })
             .map(|l| *l)
     }
 
-    pub fn load_language_at_path(&self, src_path: &Path, header_path: &Path) -> Result<Language> {
+    pub fn load_language_at_path(
+        &self,
+        src_path: &Path,
+        header_path: &Path,
+        external_files: &Option<Vec<String>>,
```

Comment:
The `external_files` parameter looks not very semantically correct for this function, I think it would be better to replace it on a `force_recompiling: bool` parameter and make this decision on https://github.com/tree-sitter/tree-sitter/pull/2545/files#diff-eb2bc03777ae5bf9ec8c213b1be1a691a778c1d60594f46bed9a7f9831dafedaR308